src/core/Form/SearchInput/SearchInput.test.tsx

```diff
@@ -199,6 +199,46 @@ describe('props', () => {
       }
     });
   });
+  describe('debounce', () => {
+    it('delays the running of onChange by the given time', () => {
+      jest.useFakeTimers();
+      const mockOnChange = jest.fn();
+      const searchInput = (
+        <SearchInput
+          labelText="Debounced search"
+          clearButtonLabel="clear"
+          searchButtonLabel="search"
+          data-testid="debounced-search"
+          debounce={1000}
+          onChange={mockOnChange}
+        />
+      );
+      const { getByTestId } = render(searchInput);
+
+      const inputElement = getByTestId('debounced-search') as HTMLInputElement;
```

Comment:
no need to change this, but getByRole, getByText etc. should be preferred over test id and other arbitrary queries. See https://testing-library.com/docs/guide-which-query

Reply:
I've now changed all the tests that can plausibly use getByRole to use that.

Reply:
Great work!